core/src/main/java/org/apache/calcite/rel/rules/FilterProjectTransposeRule.java

```diff
@@ -118,6 +121,14 @@ public void onMatch(RelOptRuleCall call) {
     final RelBuilder relBuilder = call.builder();
     RelNode newFilterRel;
     if (copyFilter) {
+      // Perform some conservative simplifications without assuming anything for unknown values.
+      RexSimplify simplifier =
+          new RexSimplify(relBuilder.getRexBuilder(), RelOptPredicateList.EMPTY, false,
```

Comment:
Please use `unknownAsFalse` here since `where 1>null` is the same as `where null` which is the same as `where false`

Reply:
You are right. I updated the PR but let's wait a bit since it seems that Julian has some doubts about the origin of the problem (see discussion on the JIRA).